pkg/build/controller/controller.go

```diff
@@ -273,6 +276,19 @@ func (bc *BuildController) resolveOutputDockerImageReference(build *buildapi.Bui
 	return ref, nil
 }
 
+func handleBuildCompletion(build *buildapi.Build, runPolicies []policy.RunPolicy) {
+	runPolicy := policy.ForBuild(build, runPolicies)
```

Comment:
looking at this code again, should we not determine if the build is complete first before getting the run policy and calling onComplete on it?